src/storage/src/engine.rs

```diff
@@ -456,6 +460,17 @@ impl<S: LogStore> EngineInner<S> {
         Ok(region)
     }
 
+    async fn drop_region(&self, name: &str, opts: &DropOptions) -> Result<()> {
+        let manifest_dir = region_manifest_dir(&opts.parent_dir, name);
+        self.object_store
+            .delete(&manifest_dir)
+            .await
+            .context(error::DeleteObjectSnafu { path: manifest_dir })?;
```

Comment:
Yep, I was a bit confused about the file structure here earlier, we can directly use `path` in `ManifestObjectStore`.
Should we delete logs in the order of their versions to prevent premature deletion of the latest logs?

Reply:
We can delete all manifest `.json` files, and the checkpoint files unless the last `.json` file (which contains the `Delete` action) in an OpenDAL batch delete op. Finally, we remove the latest `.json` file.

Reply:
I think we can assume that the last `.json` file
1. list all files (including `.json`, `.checkpoint` and `last_checkpoint`)
2. find out the file with the largest id and filter it
3. remove the remaining files via `remove(&paths)`
4. Remove the last file and the directory via `remove_all()`